tasks/gaia.js

```diff
@@ -0,0 +1,302 @@
+      },
+      {
+        denom: `localcoin`,
+        amount: `1000`
+      }
+    ]
+  })
+  fs.writeJSONSync(genesisLocation, genesis)
+
+  await makeExecWithInputs(
+    `${nodeBinary} gentx --name ${keyName} --home ${nodeHomeDir} --home-client ${clientHomeDir}`,
+    [password]
+  )
+
+  await makeExec(`${nodeBinary} collect-gentxs --home ${nodeHomeDir}`)
+
+  genesis = fs.readJSONSync(genesisLocation)
+  return genesis
+}
+
+function getGenesis(homeDir) {
+  const genesisLocation = path.join(homeDir, `config/genesis.json`)
+  let genesis = fs.readJSONSync(genesisLocation)
+  return genesis
+}
+
+// make it so that one initialized node will become a validator
+// the operator of that validator needs to declare the validator
+// therefor the operator needs funds to do initial staking
+// therefor we need to send some tokens from another account to the operator account
+async function makeValidator(
+  mainSignInfo, // main account that holds funds
+  nodeHome,
+  cliHome,
+  moniker,
+  operatorSignInfo = {
+    keyName: `local`,
+    password: `1234567890`,
+    clientHomeDir: cliHome
+  }
+) {
+  let valPubKey = await getValPubKey(nodeHome)
+  let { address: operatorAddress } = await createKey(operatorSignInfo)
+  await sendTokens(mainSignInfo, `10steak`, operatorAddress)
+  while (true) {
+    console.log(`Waiting for funds to delegate`)
+    try {
+      await sleep(1000)
+      await getBalance(cliHome, operatorAddress)
+    } catch (err) {
+      console.error(err) // kept in here to see if something unexpected fails
+      continue
+    }
+    break
+  }
+  await declareValidator(
+    operatorSignInfo, // key name that holds funds and is the same address as the operator address
+    moniker,
+    valPubKey,
+    operatorAddress
+  )
+}
+
+async function getValPubKey(node_home) {
+  let command = `${nodeBinary} tendermint show-validator --home ${node_home}`
+  const stdout = await makeExec(command)
+  return stdout.trim()
+}
+async function getNodeId(node_home) {
+  let command = `${nodeBinary} tendermint show-node-id --home ${node_home}`
+  const stdout = await makeExec(command)
+  return stdout.trim()
+}
+async function getBalance(cliHome, address) {
+  let command = `${cliBinary} query account ${address} --home ${cliHome} --output "json" --trust-node`
+  const stdout = await makeExec(command)
+  return JSON.parse(stdout.trim())
+}
+
+// sends a create-validator tx
+async function declareValidator(
+  { keyName, password, clientHomeDir }, // operatorSignInfo
+  moniker,
+  valPubKey,
+  operatorAddress
+) {
+  let command =
+    `${cliBinary} tx create-validator` +
+    ` --home ${clientHomeDir}` +
+    ` --from ${keyName}` +
+    ` --amount=10steak` +
+    ` --pubkey=${valPubKey}` +
+    ` --address-delegator=${operatorAddress}` +
+    ` --moniker=${moniker}` +
+    ` --chain-id=test_chain` +
+    ` --commission-max-change-rate=0` +
+    ` --commission-max-rate=0` +
+    ` --commission-rate=0` +
+    ` --json`
+
```

Comment:
```JavaScript
let command = `${cliBinary} tx create-validator
  --home ${clientHomeDir}
  --from ${keyName}
  --amount=10steak
  --pubkey=${valPubKey}
  --address-delegator=${operatorAddress}
  --moniker=${moniker}
  --chain-id=test_chain
  --commission-max-change-rate=0
  --commission-max-rate=0
  --commission-rate=0
  --json`
```

Reply:
this command includes several `\n` which doesn't work